### ISSUE TITLE
[Bug] Dokan: Files are not sync when using git annex.

### ISSUE THREAD

**Issue by fbewivpjsbsby** [equalitie/ouisync-app#414]
### Version, approach of installation, operating system

1. What version of ouisync are you using?
v0.7.8
2. What approach of installation do you install ouisync?
github release exe.
3. What is your operating system?
Windows 10

### Description
Files are not sync when using [git annex](https://git-annex.branchable.com).
Maybe it Copying files failed when writing in parallel.

### Steps to reproduce
1. Start ouisync and create a repository.
2. Create a git annex repository using git bash. The ouisync repository cannot be used because it does not support some metadata. It is mean create the file on any drive other than the drive where ouisync is mounted, such as C: or D:, but not O: (ouisync).
```
mkdir "C:\test"
cd "C:\test"
git init
git annex init
```
3. Set ouisync repository as git annex remote, we use 1MiB chunk.
```
git annex initremote test type=directory encryption=none directory="O:\[ouisync repository name]" chunk=1MiB
```
4. Add some files to the git annex in the ouisync repository.
```
dd if=/dev/zero of=test bs=4MiB count=1
git annex add .
git commit -m "test"
git annex sync --no-push --no-pull --no-content
git annex copy --to test
```
5. In the ouisync repository, the git annex remote looks like ```/BASE_DIR/aa/bb/FILEKEY/FILEKEY```. Open ```aa/bb```, we need to see 4 files in the directory (4MiB is 4 chunks). But sometimes we only see 1 file and other files are located in  ```/BASE_DIR/tmp/FILEKEY/FILEKEY```.
If we delete the local file, copy ouisync repository file to local, we get the error.
```
git annex drop
git annex copy --from test
```

**Comment by inetic** [comment]
Hello, thank you for creating this issue. I looked into it briefly. What seems to be going on is that `git annex copy -to test` creates a file in `/BASE_DIR/tmp/FILEKEY/FILEKEY` then calls `close` on it and later tries to move the file to `/BASE_DIR/aa/bb/FILEKEY/FILEKEY`.

What I see is that sometimes the `close` operation takes longer and does not finish before the moving from `/tmp/` to `/aa/bb/` starts. The `close` operation internally calls `flush` on the file and because it has not yet finished, the move operation may not see the file in `/tmp/` or it may not see all of the file's content.

More detailed, [this `close_file` operation](https://github.com/equalitie/ouisync/blob/d512c8e4905e5c4e0a6dc21a16ebed000589badc/vfs/src/dokan/multi_repo_mount.rs#L775-L795) does not finish before [this `move_file` operation](https://github.com/equalitie/ouisync/blob/d512c8e4905e5c4e0a6dc21a16ebed000589badc/vfs/src/dokan/multi_repo_mount.rs#L1206-L1246) starts.

It may be that I misunderstood Windows FS or Dokan's semantics in that when a file is to be re-opened (in this case in the move operation) perhaps we in our code need to ensure that the `close` operation finished.

**Comment by inetic** [comment]
@fbewivpjsbsby this should now be fixed in the latest [v0.8.0 release](https://github.com/equalitie/ouisync-app/releases/tag/v0.8.0). I'm going to close the issue, but please feel free to reopen if you still see issues.

**Comment by fbewivpjsbsby** [comment]
> @fbewivpjsbsby中得到修复 [现在，这个问题应该在最新的v0.8.0 版本 ](https://github.com/equalitie/ouisync-app/releases/tag/v0.8.0)。 我将关闭该问题，但如果您仍然发现问题，请随时重新打开。

I tested ouisync cli [5ddb344](https://github.com/equalitie/ouisync/commit/5ddb3444f889f51191f8d6ae625ed96527934807), lgtm.
But in ouisync-app 0.8.0 exe, It has error.
```
copy test (from test...)

  Verification of content failed
failed
copy: 1 failed
```
In ```/BASE_DIR/tmp``` is blank, looks good.
But in  ```/BASE_DIR/aa/bb/FILEKEY/FILEKEY```, I see all file size is 0b.

**Comment by fbewivpjsbsby** [comment]
> @fbewivpjsbsby this should now be fixed in the latest [v0.8.0 release](https://github.com/equalitie/ouisync-app/releases/tag/v0.8.0). I'm going to close the issue, but please feel free to reopen if you still see issues.

I tested ouisync cli [5ddb344](https://github.com/equalitie/ouisync/commit/5ddb3444f889f51191f8d6ae625ed96527934807), lgtm.
But in ouisync-app 0.8.0 exe, It has error.
```
copy test (from test...)

  Verification of content failed
failed
copy: 1 failed
```
In ```/BASE_DIR/tmp``` is blank, looks good.
But in  ```/BASE_DIR/aa/bb/FILEKEY/FILEKEY```, I see all file size is 0b.

**Comment by inetic** [comment]
There was a silly error and unfortunately due to our slow release process it took a lot of time to release. If it's still relevant to you and would like to give it a test, it should now be available in [v0.8.1 release](https://github.com/equalitie/ouisync-app/releases/tag/v0.8.1).

We are trying to improve the slow release and testing process and we modified the CI script to create artifacts. It can be found in [`GitHub > Actions`](https://github.com/equalitie/ouisync-app/actions)`> [Successful build] > Artifacts`

**Comment by fbewivpjsbsby** [comment]
> There was a silly error and unfortunately due to our slow release process it took a lot of time to release. If it's still relevant to you and would like to give it a test, it should now be available in [v0.8.1 release](https://github.com/equalitie/ouisync-app/releases/tag/v0.8.1).
> 
> We are trying to improve the slow release and testing process and we modified the CI script to create artifacts. It can be found in [`GitHub > Actions`](https://github.com/equalitie/ouisync-app/actions)`> [Successful build] > Artifacts`

In ouisync-app 0.8.1 exe lgtm.

**Comment by inetic** [comment]
Thanks, closing. If it re-appears, we can reopen the ticket.
